File: AwesomeProject/ios/QuickPose/QuickPoseViewManager.m
#import "React/RCTViewManager.h"

@interface RCT_EXTERN_MODULE(QuickPoseViewManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(onUpdate, RCTDirectEventBlock)
RCT_EXTERN_METHOD(updateFromManager:(nonnull NSNumber *)node)

@end

File: AwesomeProject/ios/QuickPose/QuickPoseViewManager.swift
import Foundation

@objc(QuickPoseViewManager)
class QuickPoseViewManager: RCTViewManager {
  override func view() -> UIView! {
    return QuickPoseView()
  }

  override static func requiresMainQueueSetup() -> Bool {
    return true
  }
  
  @objc func updateFromManager(_ node: NSNumber) {
    DispatchQueue.main.async {
      let component = self.bridge.uiManager.view(forReactTag: node) as! QuickPoseView
      component.update()
    }
  }
}
